README.md

```diff
@@ -42,82 +42,92 @@ Add view for your layout:
+        .code(R.string.listing_js)
+        .language("js")
+        .highlight(codeView);
 ```
-
-And perform actions sequentially when view built:
+or
 ```java
-codeView.setCodeContent(getString(R.string.listing_java));
-codeView.highlightCode("java");
+Highlighter h = new Highlighter(this)
+        .code(R.string.listing_java)
+        .shortcut(true)
+        .maxLines(10)
+        .shortcutNote("Show all")
+        .language("java");
+
+codeView.init(h);
 ```
 
-You can use both forms for build & built view, but note: ```setCodeContent(String)``` is final step when you build your view, otherwise not. If you firstly highlight and then set code content, code will not be highlighted if view was not built yet. Instructions above helps you to avoid errors. View has state to handle this behavior.
+Use CodeView.init() or use Highlighter.highlight(codeView) to init adapter and start highlighting.
 
 ## Customizing
-Use implicit form to code highlighting:
-```java
-codeView.highlightCode();
-```
-or eplixit (see available extensions below):
 ```java
-codeView.highlightCode("js"); // it will work fast!
+.language("js") // it will work fast!
+.code(content)
+.theme(ColorTheme.DEFAULT)
+.shortcut(true)
+.maxLines(6)
+.shortcutNote("Show All")
+.lineClickListener(new OnCodeLineClickListener(){})
+.shadows(true)
 ```
 
 Extend default color theme:
 ```java
 int myColor = ContextCompat.getColor(this, R.color.code_content_background);
-codeView.setColorTheme(ColorTheme.SOLARIZED_LIGHT.withBgContent(myColor));
+.theme(ColorTheme.SOLARIZED_LIGHT.withBgContent(myColor));
 ```
 or provide your own (don't forget to open PR with this stuff!)
 ```java
-codeView.setColorTheme(new ColorThemeData(new SyntaxColors(...), ...));
+.theme(new ColorThemeData(new SyntaxColors(...), ...));
 ```
 
 Handle user clicks on code lines:
 ```java
-codeView.setCodeListener(new OnCodeLineClickListener() {
+.lineClickListener(new OnCodeLineClickListener() {
     @Override
-    public void onCodeLineClicked(int n, @NotNull String line) {
+    public void onLineClicked(int n, @NotNull String line) {
         Log.i("ListingsActivity", "On " + (n + 1) + " line clicked");
     }
 });
 ```
 
-Enable shadows to hide scrolled content:
+Enable shadows to hide scrolled code:
 ```java
-codeView.setShadowsEnabled(true);
+.shadows(true);
 ```
 
 ## Adapter customization
 Sometimes you may want to add some content under line. You can create your own implementation as follows:
 
 1. Create your model to store data, for example some ```MyModel``` class.<br>
-2. Extend ```AbstractCodeAdapter<MyModel>``` typed by your model class.<br>
+2. Extend ```CodeAdapter<MyModel>``` typed by your model class.<br>
 3. Implement necessary methods in obtained ```MyCodeAdapter<MyModel>```:
 ```kotlin
 // Kotlin
-class MyCodeAdapter : AbstractCodeAdapter<MyModel> {
-    constructor(context: Context, content: String) : super(context, content)
+class MyCodeAdapter : CodeAdapter<MyModel> {
```

Comment:
Please, leave old name started with "Abstract" if **CodeAdapter** implementation is still abstract, because this makes it clear that class is not available without user logic.